src/main/java/routing/CustomHandlerRouter.java

```diff
@@ -0,0 +1,38 @@
+package com.td.HttpServer;
+import java.util.Collection;
+import java.util.ArrayList;
+public class CustomHandlerRouter implements IHandlerRouter {
+
+  private Collection<CustomRoute> routes;
+  private DefaultHandlerRouter defaultHandlerRouter;
+
+  public CustomHandlerRouter(DefaultHandlerRouter defaultHandlerRouter) {
+    this.defaultHandlerRouter = defaultHandlerRouter;
+    this.routes = new ArrayList<CustomRoute>();
+  }
+
+  public void addRoute(String method, String path, IHandler handler) {
+    CustomRoute newRoute = new CustomRoute(method, path, handler);
+    routes.add(newRoute);
+  }
+
+  public IHandler selectHandlerBadRequest() {
+    return new HandlerBadRequest();
+  }
+
+  public IHandler selectHandler(HttpRequest request) {
+    boolean found = false;
+    IHandler handler = selectHandlerBadRequest();
```

Comment:
I did this because the complier was giving me a warning that handler might not get set. Even thought it always should. Probably, not the best way to make that warning go away. I'll figure out a better way to do this.